languages/swift/exercises/concept/functions/.docs/introduction.md

```diff
@@ -0,0 +1,116 @@
+}
+
+greeting(name: "Bobo", duration: 7)
+// => "Welcome, Bobo. Enjoy your 7 night stay."
+greeting(duration: 3)
+// => "Welcome, guest. Enjoy your 3 night stay."
+greeting(name: "Wynona")
+// => "Welcome, Wynona. Enjoy your 2 night stay."
+greeting()
+// => "Welcome, guest. Enjoy your 2 night stay."
+```
+
+## Variadic parameters
+
+_Variadic parameters_ in Swift allow zero or more values of the same type to be passed into a single parameter in a function. This is indicated by appending `...` to the type annotation of the parameter.
+
+These values will be automatically grouped into an array with elements of the same type as the type of the variadic parameter.
+
+```swift
+func geometricMean(_ numbers: Double...) -> Double {
+  var total = 1.0
+  for number in numbers {
+      total *= number
+  }
+  return pow(total, 1.0 / Double(numbers.count))
+}
+
+geometricMean(1, 2, 3, 4, 5)
+// => 2.605171084697352
+```
+
+## In-out parameters
+
+Within the body of a function, parameters are treated as constants, not variables; trying to modify the value of a parameter will raise a compile-time error. If a function wishes to modify the value of a parameter, it must use an _in-out parameter_ to make this mutability explicit.
+
+To use in-out parameters, a programmer must be sure of three things:
+
+1. The function definition must explicitly mark in-out parameters by writing the keyword `inout` between the `:` and the type name in the parameter's type annotation.
+2. The value passed in to the in-out parameter must be declared as a variable, not a constant. Similarly it cannot be a literal.
+3. The variable name must be preceded by an `&` in the call to the function, explicitly marking it as modifiable.
+
+When this is done, the function will make an internal copy of the in-out parameter, modify it as directed in the body of the function and then copy the modified value back into the variable that was passed in when the function returns.
+
+```swift
+func updateVersion(_ record: inout (version: Int, title: String)) {
+  record.version += 1
+}
+
+var dbRecord = (version: 2, title: "Exercism")
+updateVersion(&dbRecord)
+
+// dbRecord is now (3, "Exercism")
+```
+
+## Nested functions
+
+Functions may be defined inside of other functions. This is commonly used to create helper functions which are only useful to their enclosing function and so don't need to pollute the outside namespace.
+
+These functions are defined and called just like normal functions, but are not visible outside the enclosing function.
+
+```swift
+func makePangram(_ str: String) -> String {
+  func missingLetters() -> String {
+    let base = str.lowercased() + "isay"
+    var missing = ""
+    for char in "abcdefghijklmnopqrstuvwxyz" {
+      if char.isLetter && !base.contains(char) {
+        missing.append(char)
+      }
+    }
+    print("missing:", missing)
+    return missing
+  }
+  return "\(str) I say, \"\(missingLetters())\""
+}
```

Comment:
This example is quite big, which can make it somewhat harder for the student to quickly grasp what it happening. Maybe you could come up with a simpler example?

Reply:
Let me see what I can come up with.